corehq/apps/hqwebapp/management/commands/build_requirejs.py

```diff
@@ -37,251 +35,261 @@ def add_arguments(self, parser):
+                # src is something like .../staticfiles/foo/baz/bar.js, so search local_js_dirs
+                # for something ending in foo/baz
+                common_dir = os.path.relpath(os.path.dirname(src), self._staticfiles_path())
+                options = [d for d in local_js_dirs if re.search(r'\b' + re.escape(common_dir), d)]
+                if len(options) == 1:
+                    dest_stem = options[0][:-len(common_dir)]   # trim the common foo/baz off the destination
+                    copyfile(src, os.path.join(settings.BASE_DIR, dest_stem, module['name'] + '.js'))
+                else:
+                    logger.warning("Could not copy {} to {}".format(os.path.relpath(src), os.path.relpath(dest)))
+
+    def _update_resource_hash(self, name, filename):
+        file_hash = self.get_hash(filename)
+        self.resource_versions[name] = file_hash
+        return file_hash
+
+    # Overwrite source map references. Source maps are accessed on the CDN, so they need the version hash
+    def _update_source_maps(self, config):
+        if not self.optimize:
+            return
+
+        logger.info("Updating resource_versions with hashes from newly minified bundles")
 
-        # Overwrite each bundle in resource_versions with the sha from the optimized version in staticfiles
         for module in config['modules']:
-            filename = os.path.join(ROOT_DIR, 'staticfiles', module['name'] + ".js")
-
-            # TODO: it'd be a performance improvement to do this after the `open` below
-            # and pass in the file contents, since get_hash does another read.
-            file_hash = self.get_hash(filename)
-
-            if optimize:
-                # Overwrite source map reference. Source maps are accessed on the CDN,
-                # so they need to have the version hash appended.
-                with open(filename, 'r') as fin:
-                    lines = fin.readlines()
-                with open(filename, 'w') as fout:
-                    for line in lines:
-                        if re.search(r'sourceMappingURL=bundle.js.map$', line):
-                            line = re.sub(r'bundle.js.map', 'bundle.js.map?version=' + file_hash, line)
-                        fout.write(line)
-            resource_versions[module['name'] + ".js"] = file_hash
-
-        # Write out resource_versions.js for all js files in resource_versions
-        # Exclude formdesigner directory, which contains a ton of files, none of which are required by HQ
-        filename = os.path.join(ROOT_DIR, 'staticfiles', 'hqwebapp', 'js', 'resource_versions.js')
+            filename = self._staticfiles_path(module['name'] + ".js")
+            with open(filename, 'r') as fin:
+                lines = fin.readlines()
+            with open(filename, 'w') as fout:
+                for line in lines:
+                    match = re.search(r'sourceMappingURL=(bundle.b[35].js.map)$', line)
```

Comment:
What do you think of moving this regex to a constant at the top of the file?
```py
BUNDLE_MAP_PATTERN = re.compile(r'sourceMappingURL=(bundle.b[%s].js.map)$'
                                % "".join(str(v) for v in BOOTSTRAP_VERSIONS))
```
Then this line would change to
```suggestion
                    match = BUNDLE_MAP_PATTERN.search(line)
```

Reply:
58dafa59cfd - this also combines the requirejs_main regexes into a single constant, which is a nice simplification

Reply:
Followup in 4048d572c89 

Annoyingly and surprisingly, I couldn't get the same regex working in both grep and python, needed to do more escaping for grep. But I also noticed that the old grep expression was only capturing `requirejs_main`, not `requirejs_main_b5`, which was a bug.

Also made the versions in the regex dynamic, which I missed when first reading your comment.